### DISCUSSION TITLE
Issue with passing variables -> Multi Select

### DISCUSSION THREAD

**Discussion by VikingDadMedic** [Budibase/budibase#5224] [Help]
Could someone explain what I am doing incorrectly, I am trying to **Create a New Row** using some json data - however, no matter what I seem to do, I can't populate the **multi-select** fields, using the _json data_, from the **automate** steps.

**Data**
`{
    "Destinations":[
        "North America",
        "Asia",
        "South America",
        "Australia/Oceania"
    ]
}`

**Desired Outcome**
<img width="1007" alt="Screen Shot 2022-04-04 at 12 38 55 AM" src="https://user-images.githubusercontent.com/91684138/161475269-41c2f495-bb32-4b89-ba54-6ce0ac806019.png">

**Error Message**
<img width="443" alt="Screen Shot 2022-04-04 at 12 32 38 AM" src="https://user-images.githubusercontent.com/91684138/161474447-af26dfff-5323-45a2-a2b0-ff54df5d24f5.png">

**Comment by melohagan** [comment]
Hi @VikingDadMedic 

What does your trigger look like (*When this happens*)? Would you be able to send a screenshot please?

**Reply by VikingDevMedic** [reply]
![75FA5D7D-994D-4531-BFA2-BCD3D6E981FD](https://user-images.githubusercontent.com/101609169/161669104-451a7fd5-da20-4aa8-b4e6-04862503351b.jpeg)
![D5985536-9FE8-47D0-96AF-8F2D95C7BEA5](https://user-images.githubusercontent.com/101609169/161669105-6ebda3ed-4635-405b-98a2-c0f3bbe0ceba.jpeg)
![4C58ECDE-D46A-4AC5-B8D8-268095935F56](https://user-images.githubusercontent.com/101609169/161669106-7240d5a7-ab95-46cb-9907-f346138377f6.jpeg)

**Reply by melohagan** [reply]
So when you setup the webhook schema, what payload did you POST?

**Reply by VikingDadMedic** [reply]
See below

<img width="733" alt="schema-post" src="https://user-images.githubusercontent.com/91684138/161767104-b8ec701b-c548-410b-90c5-50edb3a48f99.png">

**Reply by melohagan** [reply]
Could you also click the edit pencil on your multi-select column and send through a screenshot please. Just want to double check that the options from the schema are also listed in the multi-select column options.

**Reply by VikingDadMedic** [reply]
<img width="1041" alt="table_column_schema" src="https://user-images.githubusercontent.com/91684138/161802505-87d75cba-631c-4e37-b921-2d592ab5aced.png">

**Comment by melohagan** [comment]
Vikings, 

Thank you for the screenshots! 

I have been able to recreate your issue, and I noticed that it does pass for a test array with a single element. 

```
"Destinations": [
      "North America"
    ]
```

It may be the case that it is just the validation that's off. 
If you open your published app, and then send down a payload via your trigger URL with multiple destinations, does it behave as expected?

**Reply by VikingDadMedic** [reply]
Negative. 

However this works: 
`{{trigger.Destinations.[0]}}, `
`{{trigger.Destinations.[1]}}, `
`{{trigger.Destinations.[2]}}, `
`{{trigger.Destinations.[3]}}, `
`{{trigger.Destinations.[4]}}, `
`{{trigger.Destinations.[5]}}, `
`{{trigger.Destinations.[6]}}, `
`{{trigger.Destinations.[7]}}, `
`{{trigger.Destinations.[8]}}, `
`{{trigger.Destinations.[9]}} `

BUT

there appears to be an issue with how BB processes a ` , `.

So if you have any idea how to iterate in BB like the `{{#forEach}}` - would love to know how exactly to refactor the above  **handlebars style syntax**.  I say this because I have no issue having iterating in the handlebars repl - however, the same code produces side effects in BB.

**Reply by melohagan** [reply]
ForEach functionality for automations is being worked on, and will hopefully be available in the next release: https://github.com/Budibase/budibase/issues/4427 and https://github.com/Budibase/budibase/issues/4551

I have also raised a bug here: https://github.com/Budibase/budibase/issues/5346

A potential work around in the meantime could be to use a JSON column instead of a multi-select column. It could hold an array of values.
